aggregator_core/src/datastore.rs

```diff
@@ -2537,6 +2537,44 @@ ORDER BY id DESC
         )
     }
 
+    /// Returns the minimal time interval spanned by the timestamps of the reports included in the
+    /// collection, or `None` if there are no included reports.
+    #[tracing::instrument(skip(self), err)]
+    pub async fn get_interval_spanned_by_collection(
+        &self,
+        collection_job_id: &CollectionJobId,
+    ) -> Result<Option<Interval>, Error> {
+        let stmt = self
+            .prepare_cached(
+                "SELECT aggregation_jobs.client_timestamp_interval FROM collection_jobs
+                JOIN aggregation_jobs
+                ON aggregation_jobs.task_id = collection_jobs.task_id
+                AND (aggregation_jobs.batch_id = collection_jobs.batch_identifier
+                    OR aggregation_jobs.client_timestamp_interval && collection_jobs.batch_interval)
+                WHERE collection_jobs.collection_job_id = $1",
```

Comment:
We should probably have separate queries/methods to find this for time interval and fixed size tasks. This logic works for fixed size tasks, because each aggregation job corresponds to a single batch, and we can directly use `aggregation_jobs.client_timestamp_interval`. However, with time interval queries, there's no correspondence between aggregation jobs and batches. We'd have to instead look up the minimum and maximum timestamp of aggregated reports within the query's batch interval.

Reply:
Bonus: we could aggregate intervals for the fixed-size query with `SELECT tsrange(min(lower(aggregation_jobs.client_timestamp_interval)), max(upper(aggregation_jobs.client_timestamp_interval)), '[)')`. This would save us having to transfer intermediate  values, and may allow a better query plan by pushing down the aggregate.

Reply:
I don't think I follow. This is the same logic used by `delete_expired_aggregation_artifacts` to determine if an aggregation job belongs to a collection, and the `collection_interval_spanned` I added shows that we can correctly determine which aggregation jobs belong to which collections in a time interval task. What case is not covered by this?

Reply:
Assume we have four reports, two at $T_1$, and two at $T_2$, two collections, one covering only $T_1$, and one covering $T_2$, and two aggregation jobs. If both aggregation jobs contain one report at $T_1$ and one report at $T_2$, then the `client_timestamp_interval` for both aggregation jobs will span both times, and this query will report that each collection's covered interval is longer than the batch interval, when it should only ever be equal to the batch interval or shorter.

Reply:
I think I see now, thank you. I have to think about this more (after lunch), but I think it's already incidentally true that every time interval aggregation job we schedule belongs to a single collection. However I need to take a harder look at how we create aggregation jobs, batch aggregations and collections.

Reply:
I believe that changed in #967, as we no longer build a group map by rounded report timestamp, and instead just chunk up unordered reports.

Reply:
Ah hah, that'll do it. OK, so we're back to the drawing board on this one. I suspect we have a similar problem with the garbage collection query.